compute/main.go

```diff
@@ -0,0 +1,599 @@
+
+	_, err = deleteNetWorkInterface(ctx, conn)
+	if err != nil {
+		log.Fatalf("cannot delete network interface:%+v", err)
+	}
+	log.Println("deleted network interface")
+
+	_, err = deleteNetworkSecurityGroup(ctx, conn, nsgName)
+	if err != nil {
+		log.Fatalf("cannot delete network security group:%+v", err)
+	}
+	log.Println("deleted network security group")
+
+	_, err = deletePublicIP(ctx, conn, publicIPName)
+	if err != nil {
+		log.Fatalf("cannot delete public IP address:%+v", err)
+	}
+	log.Println("deleted public IP address")
+
+	_, err = deleteSubnets(ctx, conn, vnetName, subnetName)
+	if err != nil {
+		log.Fatalf("cannot delete subnet:%+v", err)
+	}
+	log.Println("deleted subnet")
+
+	_, err = deleteVirtualNetWork(ctx, conn)
+	if err != nil {
+		log.Fatalf("cannot delete virtual network:%+v", err)
+	}
+	log.Println("deleted virtual network")
+
+	_, err = deleteResourceGroup(ctx, conn)
+	if err != nil {
+		log.Fatalf("cannot delete resource group:%+v", err)
+	}
+	log.Println("deleted resource group")
+	log.Println("success deleted virtual machine.")
+}
+
+func connectionAzure() (*armcore.Connection, error) {
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	conn := armcore.NewDefaultConnection(cred, &armcore.ConnectionOptions{
+		Logging: azcore.LogOptions{
+			IncludeBody: true,
+		},
+	})
+	return conn, nil
+}
+
+func createResourceGroup(ctx context.Context, connection *armcore.Connection) (*armresources.ResourceGroup, error) {
+	resourceGroup := armresources.NewResourceGroupsClient(connection, subscriptionId)
+
+	parameters := armresources.ResourceGroup{
+		Location: to.StringPtr(location),
+		Tags:     map[string]*string{"sample-rs-tag": to.StringPtr("sample-tag")}, // resource group update tags
+	}
+
+	resp, err := resourceGroup.CreateOrUpdate(ctx, resourceGroupName, parameters, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.ResourceGroup, nil
+}
+
+func deleteResourceGroup(ctx context.Context, connection *armcore.Connection) (*http.Response, error) {
+	resourceGroup := armresources.NewResourceGroupsClient(connection, subscriptionId)
+
+	pollerResponse, err := resourceGroup.BeginDelete(ctx, resourceGroupName, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func createVirtualNetwork(ctx context.Context, connection *armcore.Connection) (*armnetwork.VirtualNetwork, error) {
+	vnetClient := armnetwork.NewVirtualNetworksClient(connection, subscriptionId)
+
+	parameters := armnetwork.VirtualNetwork{
+		Resource: armnetwork.Resource{
+			Location: to.StringPtr(location),
+		},
+		Properties: &armnetwork.VirtualNetworkPropertiesFormat{
+			AddressSpace: &armnetwork.AddressSpace{
+				AddressPrefixes: []*string{
+					to.StringPtr("10.1.0.0/16"), // example 10.1.0.0/16
+				},
+			},
+			//Subnets: []*armnetwork.Subnet{
```

Comment:
why commented out?

Reply:
Because I have a createSubnet behind, I don't want to create two subnets in this sample, here is just another way to create a subnet.